okhttp/src/main/kotlin/okhttp3/internal/http/CallServerInterceptor.kt

```diff
@@ -37,8 +37,11 @@ class CallServerInterceptor(private val forWebSocket: Boolean) : Interceptor {
     var invokeStartEvent = true
     var responseBuilder: Response.Builder? = null
     var sendRequestException: IOException? = null
+
+    // Send request headers, or if failed then avoid any attempt to read an early response.
+    exchange.writeRequestHeaders(request)
```

Comment:
I'm not happy with the possibly redundant code in the catch block below. cc @swankjesse